src/lib/viewers/controls/annotations/AnnotationsControls.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import noop from 'lodash/noop';
+import AnnotationsButton from './AnnotationsButton';
+import IconHighlightText16 from '../icons/IconHighlightText16';
+import IconRegion24 from '../icons/IconRegion24';
+import useFullscreen from '../hooks/useFullscreen';
+import { AnnotationMode } from './types';
+import './AnnotationsControls.scss';
+
+export type Props = {
+    annotationMode?: AnnotationMode;
+    fileId: string;
+    onAnnotationModeClick?: ({ mode }: { mode: AnnotationMode }) => void;
+    hasHighlight?: boolean;
+    hasRegion?: boolean;
+};
+
+export default function AnnotationsControls({
+    annotationMode = AnnotationMode.NONE,
+    fileId,
+    onAnnotationModeClick = noop,
+    hasHighlight = true,
+    hasRegion = true,
+}: Props): JSX.Element | null {
+    const isFullscreen = useFullscreen();
+    const showHighlight = !isFullscreen && hasHighlight;
+    const showRegion = !isFullscreen && hasRegion;
+
+    if (!showHighlight && !showRegion) {
+        return null;
+    }
+
+    const handleModeClick = (mode: AnnotationMode): void => {
+        onAnnotationModeClick({ mode: annotationMode === mode ? AnnotationMode.NONE : mode });
+    };
+
+    return (
+        <div className="bp-AnnotationsControls">
+            <AnnotationsButton
+                data-resin-target="highlightRegion"
+                data-testid="bp-AnnotationsControls-regionBtn"
+                fileId={fileId}
+                isActive={annotationMode === AnnotationMode.REGION}
+                isEnabled={showRegion}
+                mode={AnnotationMode.REGION}
```

Comment:
Instead of having to pass the `mode` as a prop into each button, what if we just declare that in the callback logic in this component? We could have separate callbacks like `handleRegionClick` or wrap `handleModeClick` in a closure with the mode?

Reply:
I figure there will be more modes and buttons coming in the future, so this cuts down on the repetitive callbacks a bit.